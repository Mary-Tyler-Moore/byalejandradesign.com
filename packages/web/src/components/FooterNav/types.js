/** @flow */
export type Props = {
  footerNav: Array<{
    link: string,
    label: string,
  }>,
};
