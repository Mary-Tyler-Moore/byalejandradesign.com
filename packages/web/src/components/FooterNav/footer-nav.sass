@import ../../sass/global.sass

.footerNav
  width: 100%
  display: flex
  justify-content: space-between

  &_link
    @extend %navLink

    color: $white
    margin: 0
    padding: 1rem 0
    margin: 0 1.5rem

    @extend %subtitle-copy-1

    +all-phone
      margin: 0
      padding: 0


    &:hover
      @extend %navHighlight

    &:first-child
      margin-left: 0

    &:last-child
      margin-right: 0
