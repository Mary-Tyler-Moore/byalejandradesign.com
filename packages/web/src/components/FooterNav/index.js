/** @flow */
export { default } from './FooterNav'
export * from './types'

