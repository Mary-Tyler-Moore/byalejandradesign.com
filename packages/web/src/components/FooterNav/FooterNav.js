/** @flow */
import * as React from 'react';
import { Link } from 'gatsby';
import URLToTitle from '../Nav/url-to-title';
// types
import type { Props } from './types';
// styles
import './footer-nav.sass';

const FooterNav = (props: Props) => (
  <nav className="footerNav">
    {props.footerNav.map(({ link, label }) => (
      <React.Fragment>
        <Link
          key={link}
          to={link}
          className="footerNav_link"
          activeClassName="footerNav_link-active"
        >
          {label}
        </Link>
      </React.Fragment>
    ))}
  </nav>
);

export default FooterNav;
