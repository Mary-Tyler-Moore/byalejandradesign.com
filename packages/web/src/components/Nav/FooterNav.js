import * as React from 'react';
import { Link } from 'gatsby';
import { Form } from '@njmyers/component-library';
import Button from '../Button';
import URLToTitle from './url-to-title';
// style
import './footer-nav.sass';

class DesktopFooterNav extends React.Component {
  onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();

    console.log('clicked');
  };

  render() {
    return (
      <footer className="footer">
        <section className="footer_widgets">
          <form
            name="emailSignup"
            className="emailSignup"
            onSubmit={this.onSubmit}
          >
            <Form.Input
              block="emailSignup"
              name="email"
              label="Signup For Our Newsletter"
              required
            />
            <Button type="submit">Sign Up</Button>
          </form>
        </section>
        <nav className="footerNavDesktop">
          {this.props.footerNav.map((nav) => (
            <Link
              key={nav}
              to={nav}
              className="footerNavDesktop_link"
              activeClassName="footerNavDesktop_link-active"
            >
              {URLToTitle(nav)}
            </Link>
          ))}
        </nav>
      </footer>
    );
  }
}

class MobileFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavMobile">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavMobile_link"
            activeClassName="footerNavMobile_link-active"
            onClick={this.props.onClick}
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

const FooterNav = (props) =>
  props.sizes.mobile ? (
    <MobileFooterNav {...props} />
  ) : (
    <DesktopFooterNav {...props} />
  );

export default FooterNav;
