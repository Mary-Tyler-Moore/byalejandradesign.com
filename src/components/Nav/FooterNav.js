import * as React from 'react';
import Link from 'gatsby-link';
import URLToTitle from './url-to-title';
//
import './footer-nav.sass';

class DesktopFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavDesktop">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavDesktop_link"
            activeClassName="footerNavDesktop_link-active"
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

class MobileFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavMobile">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavMobile_link"
            activeClassName="footerNavMobile_link-active"
            onClick={this.props.onClick}
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

const FooterNav = (props) =>
  props.sizes.mobile ? (
    <MobileFooterNav {...props} />
  ) : (
    <DesktopFooterNav {...props} />
  );

export default FooterNav;
