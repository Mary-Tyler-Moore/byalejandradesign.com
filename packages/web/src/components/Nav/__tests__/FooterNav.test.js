import React from 'react';
import FooterNav from '../FooterNav';
import { MemoryRouter } from 'react-router-dom';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const footerNav = ['/link', '/other-link'];

window.__PATH_PREFIX__ = '';

describe('it renders the footer navs', () => {
  test('it displays the desktop version', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <FooterNav sizes={{ mobile: false }} footerNav={footerNav} />
      </MemoryRouter>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('it displays the mobile version', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <FooterNav sizes={{ mobile: true }} footerNav={footerNav} />
      </MemoryRouter>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
