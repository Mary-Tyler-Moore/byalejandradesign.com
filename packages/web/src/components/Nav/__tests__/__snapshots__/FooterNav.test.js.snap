// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the footer navs it displays the desktop version 1`] = `"<footer class=\\"footer\\"><section class=\\"footer_widgets\\"><form name=\\"emailSignup\\" class=\\"emailSignup\\" method=\\"POST\\" data-netlify=\\"true\\"><div class=\\"emailSignup\\"><label element=\\"label\\" for=\\"email\\" class=\\"emailSignup_label\\">Signup For Our Newsletter</label><input type=\\"text\\" element=\\"input\\" name=\\"email\\" required=\\"\\" placeholder=\\"\\" class=\\"emailSignup_input\\"/></div><div class=\\"formInput\\"><input type=\\"submit\\" element=\\"input\\" value=\\"Sign Up\\" class=\\"formInput_input\\"/></div></form></section><nav class=\\"footerNavDesktop\\"><a class=\\"footerNavDesktop_link\\" href=\\"/link\\">Link</a><a class=\\"footerNavDesktop_link\\" href=\\"/other-link\\">Other Link</a></nav></footer>"`;

exports[`it renders the footer navs it displays the mobile version 1`] = `"<nav class=\\"footerNavMobile\\"><a class=\\"footerNavMobile_link\\" href=\\"/link\\">Link</a><a class=\\"footerNavMobile_link\\" href=\\"/other-link\\">Other Link</a></nav>"`;
